src/qt/pivx/mnmodel.cpp

```diff
@@ -22,17 +22,13 @@ void MNModel::updateMNList()
     int end = nodes.size();
     nodes.clear();
     collateralTxAccepted.clear();
-    for (CMasternodeConfig::CMasternodeEntry mne : masternodeConfig.getEntries()) {
+    for (const CMasternodeConfig::CMasternodeEntry& mne : masternodeConfig.getEntries()) {
         int nIndex;
         if (!mne.castOutputIndex(nIndex))
             continue;
         uint256 txHash(mne.getTxHash());
         CTxIn txIn(txHash, uint32_t(nIndex));
         CMasternode* pmn = mnodeman.Find(txIn.prevout);
-        if (!pmn) {
-            pmn = new CMasternode();
-            pmn->vin = txIn;
-        }
```

Comment:
need to keep this for MNs that have been removed from the mnodeman due to ping expiry.

Reply:
Isn't a bug, it's the expected behavior.

Masternodes who are not longer in the network list (or who have never been in the list) are presented as "MISSING".
It hasn't been changed on this PR, check the previous code here cfbf0b8b2491665b09163b353ff7480f9bfccecb . It's fixed to "MISSING" always.

Can think on additions to state more clearly the reason and not only go with "MISSING" state in a future PR for sure but for now, this is merely fixing an introduced bug.

Reply:
Aside from that, and after a briefly talk with zebra, it's true that need to re add the vin set, otherwise the txid will not be shown. Good 👍 .
Will work on a wrapper so we can store the data without need to create an empty masternode. It will still be shown as missing as before with the correct information.